nemo_skills/inference/server/model.py

```diff
@@ -552,16 +553,28 @@ def __init__(
             base_url = os.getenv("NEMO_SKILLS_OPENAI_BASE_URL", f"http://{host}:{port}/v1")
 
         if api_key is None:
-            if base_url is not None and 'api.nvidia.com' in base_url:
-                api_key = os.getenv("NVIDIA_API_KEY", api_key)
-                if not api_key:
-                    raise ValueError("NVIDIA_API_KEY is required for Nvidia-hosted models.")
+            if base_url is not None and 'nvidia.com' in base_url:
+                if endpoint_type == "azureopenai":
```

Comment:
can nvidia models be accessed through azure? Can you give an example of usage when this condition is triggered? Or did you mean to put it as a separate if, not inside `'if 'nvidia.com' in base_url`?